checks/apps/cp2k/cp2k_check.py

```diff
@@ -1,151 +1,230 @@
+    descr = "Fetch CP2K source code"
+    sourcedir = None
+    executable = "wget"
+    executable_opts = [
+        f"https://github.com/cp2k/cp2k/archive/refs/tags/v{version}.tar.gz",
+    ]
+    local = True
+
+    @sanity_function
+    def validate_download(self):
+        return sn.assert_eq(self.job.exitcode, 0)
+
+
+@rfm.simple_test
+class Cp2kBuildTest(rfm.CompileOnlyRegressionTest):
+    """
+    Test CP2K build from source.
+    """
+
+    descr = "CP2K Build Test"
+    valid_systems = ["*"]
+    valid_prog_environs = ["+cp2k-dev"]
+    build_system = "CMake"
+    sourcesdir = None
+    maintainers = ["SSA"]
+    cp2k_sources = fixture(cp2k_download, scope="session")
+    build_locally = False
+    tags = {"uenv"}
+
+    @run_before("compile")
+    def prepare_build(self):
+        self.uarch = uenv.uarch(self.current_partition)
+        self.build_system.builddir = os.path.join(self.stagedir, "build")
+        self.skip_if_no_procinfo()
+        cpu = self.current_partition.processor
+        self.build_system.max_concurrency = cpu.info["num_cpus_per_socket"]
+
+        tarsource = os.path.join(
+            self.cp2k_sources.stagedir, f"v{self.cp2k_sources.version}.tar.gz"
+        )
+
+        # Extract source code
+        self.prebuild_cmds = [
+            f"tar --strip-components=1 -xzf {tarsource} -C {self.stagedir}"
+        ]
+
+        # TODO: Use Ninja generator
+        self.build_system.config_opts = [
+            "-DCP2K_ENABLE_REGTESTS=ON",  # Puts executables under exe/local_cuda/
+            "-DCP2K_USE_LIBXC=ON",
+            "-DCP2K_USE_LIBINT2=ON",
+            "-DCP2K_USE_SPGLIB=ON",
+            "-DCP2K_USE_ELPA=ON",
+            "-DCP2K_USE_SPLA=ON",
+            "-DCP2K_USE_SIRIUS=ON",
+            "-DCP2K_USE_COSMA=ON",
+            "-DCP2K_USE_PLUMED=ON",
+        ]
+
+        if self.uarch == "gh200":
+            self.build_system.config_opts += [
+                "-DCP2K_USE_ACCEL=CUDA",
+                "-DCP2K_WITH_GPU=H100",
+            ]
+
+    @sanity_function
+    def validate_test(self):
+        # INFO: Executables are in exe/FOLDER because -DCP2K_ENABLE_REGTEST=ON
+        # INFO: With -DCP2K_ENABLE_REGTEST=OFF, executables are in build/bin/ as expected
+        folder = "local_cuda" if self.uarch == "gh200" else "local"
+        executable = os.path.join(self.stagedir, "exe", folder, "cp2k.psmp")
+        return os.path.isfile(executable)
+
+
+class Cp2kCheck(rfm.RunOnlyRegressionTest):
+    executable = "./mps-wrapper.sh cp2k.psmp"
+    maintainers = ["SSA"]
+    valid_systems = ["*"]
+    valid_prog_environs = ["+cp2k"]
+
+    @run_before("run")
+    def prepare_run(self):
+        self.uarch = uenv.uarch(self.current_partition)
+
+        config = slurm_config[self.test_name][self.uarch]
+
+        # SBATCH options
+        self.job.options = [
+            f"--nodes={config['nodes']}",
+            "--ntasks-per-core=1",
+            "--partition=debug",
```

Comment:
This is not portable so I would remove it and call reframe with `-J partition=debug` (or add it in the config file)

Reply:
Yes, this is for me to test locally. I'll remove it before merging.

Reply:
Done, I added the debug partition to the configuration and removed any reference to the system from the tests. I then used `--system todi:debug` run the tests.